ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -0,0 +1,164 @@
+        )
+        if "error" in ret.lower():
+            # actual error gets printed during "execute_command"
+            raise Exception("Failed to clone the given repository")
+    elif not os.path.exists(test_repo):
+        raise FileNotFoundError("local path {} was not found".format(test_repo))
+    else:
+        print("Using local repo ", test_repo)
+        repo_local_path = test_repo
+    return repo_local_path
+
+
+def checkout_repository(ref):
+    """
+    Checkouts the repository at current directory to the given branch/commit ($ref)
+    """
+    ret = execute_command("git checkout {}".format(ref))
+    if "error" in ret.lower():
+        # actual error gets printed during "execute_command"
+        raise Exception("Failed to checkout to the given branch/commit {}".format(ref))
+    ret = execute_command("git checkout")
+    print(ret)
+
+
+def extract_test_cases_from_ref(repo_local_path, ref):
+    """
+    Navigate to the $test_repo directory, checkouts the target branch/commit ($ref) and extracts
+    the test case titles leveraging RobotFramework TestSuiteBuilder() and TestCasesFinder() classes
+    """
+    curr_dir = os.getcwd()
+    try:
+        os.chdir(repo_local_path)
+        checkout_repository(ref)
+        builder = TestSuiteBuilder()
+        testsuite = builder.build("ods_ci/tests/")
+        finder = TestCasesFinder()
+        tests = []
+        testsuite.visit(finder)
+        for test in finder.tests:
+            # print (f'"{test.tags}"') # for future reference in order to fetch test tags
+            tests.append(test.name)
+    except Exception as err:
+        print(err)
+        os.chdir(curr_dir)
+        raise
+    os.chdir(curr_dir)
+    return tests
+
+
+def generate_rf_argument_file(tests, output_filepath):
+    """
+    Writes the RobotFramework argument file containing the test selection args
+    to include the extracted new tests in previous stage of this script.
+    """
+    content = ""
+    for testname in tests:
+        content += "--test {}\n".format(testname.strip())
+    try:
+        with open(output_filepath, "w") as argfile:
+            argfile.write(content)
+    except Exception as err:
+        print("Failed to generate argument file")
+        print(err)
+
+
+def extract_new_test_cases(test_repo, ref_1, ref_2, output_argument_file):
+    """
+    Wrapping function for all the new tests extraction stages.
+    """
+    repo_local_path = get_repository(test_repo)
+    print("\n---| Extracting test cases from {} branch/commit |---".format(ref_1))
+    tests_1 = extract_test_cases_from_ref(repo_local_path, ref_1)
+    print("\nDone. Found {num} test cases".format(num=len(tests_1)))
+    print("\n---| Extracting test cases from {} branch/commit |---".format(ref_2))
+    tests_2 = extract_test_cases_from_ref(repo_local_path, ref_2)
+    print("Done. Found {num} test cases".format(num=len(tests_2)))
```

Comment:
I think having a method to wrap up these lines doesn't bring much more value here. If in the future we'll need to call the extract_test_cases_from_ref function more than 2 times, then maybe it can start bringing more benefits

Reply:
It is not critical here because it is only 2 lines.
But FMPOV as a code reviewer when I see a duplication of code:
1. For me there are more lines of code to review :) 
2. The same comments that I added before, I need to add them again.
3. code maintenance will be more complicated in the future.

Reply:
> For me there are more lines of code to review :)


Adding a new function maybe would have consisted in more than 2 new lines to review xD

Apart from that, I agree with you on the rest